design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,174 @@
+# Design Proposal: Project Deployment with Kubernetes Operator
+
+**Author**: @kfirtoledo
+
+**Begin Design Discussion**: 2023-11-30
+
+**Status:** draft
+
+## Summary/Abstract
+
+This design proposal outlines the deployment process for the ClusterLink project on a Kubernetes (K8s) cluster.  
+The proposal specifically addresses deployment within the K8s environment.\
+It will rely on a dedicated K8s operator, to make the deployment more user-friendly, simplified and native to K8s users.
+
+## Background
+
+### Motivation and problem space
+
+This design proposal aims to simplify and enhance the deployment of the ClusterLink project.
+By leveraging K8s native mechanisms, such as operators and CRDs, the objective is to establish a user-friendly deployment process.\
+Currently, ClusterLink deployment involves using the cl-adm CLI, which generates a large YAML file that contains the control plane and dataplane deployments, services, and RBAC configurations.\
+In the new design, this process will be managed by the ClusterLink operator, which will consume a simpler and more concise YAML configuration.
+
+### Impact and desired outcome
+
+Implementing this proposal is expected to improve the accessibility and ease of use for ClusterLink users.\
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management.
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+The purpose of ClusterLink is to interconnect various clusters within the same fabric.\
```

Comment:
Start with "As a site administrator I would like to join my cluster to an existing fabric" and then add the information written as additional context?